doc/user-guides/acrn-shell.rst

```diff
@@ -50,3 +50,179 @@ The ACRN hypervisor shell supports the following commands:
+
+vm_list
+=======
+
+"vm_list" provides the name of each virtual machine and its corresponding ID and
+state.
+
+.. figure:: images/shell_image8.png
+   :align: center
+
+   vm_list information
+
+vcpu_list
+=========
+
+"vcpu_list" provides information about virtual CPUs (vCPU), including
+the VM ID, PCPU ID, VCPU ID, VCPU ROLE (primary or secondary), and VCPU
+STATE (init, paused, running, zombie or unknown).
+
+.. figure:: images/shell_image7.png
+   :align: center
+
+   vcpu_list information
+
+vcpu_dumpreg
+============
+
+"vcpu_dumpreg vmid cpuid" provides vCPU related information such as
+registers values, etc.
+
+In the following example, we dump vCPU0 RIP register value and get into
+the SOS to search for the currently running function, using these
+commands::
+
+   cat /proc/kallsyms | grep RIP_value
+
+As you can see, vCPU0 is running in
+function ``acpi_idle_do_entry``.
+
+.. figure:: images/shell_image10.png
+   :align: center
+
+   vcpu_dumpreg information
+
+.. figure:: images/shell_image9.png
+   :align: center
+
+   system map information
+
+dumpmem
+=======
+
+"dumpmem mem_address" provides the specified memory target data such as
+the physical CPU (pCPU) number, etc.
+
+In this example, we know the pCPU active bitmap and physical CPU number
+physical memory address through
+``out/hv/apl-mrb-sbl/hypervisor/acrn.map``. (Note that the path for
+``acrn.map`` depends on how we build the hypervisor.)
+
+Then we can dump the memory address of the pCPU active bitmap and CPU
+number, we will know that pCPU active bitmap is 0x000000000000000f and
+pCPU number is 0x0000000000000004.
+
+.. figure:: images/shell_image12.png
+   :align: center
+
+   dumpmem information
+
+.. figure:: images/shell_image11.png
+   :align: center
+
+   acrn map information
+
+sos_console
+===========
+
+This command switches the ACRN's console to become the SOS's console.
+Use a :kbd:`Ctrl-Spacebar` to return to the ACRN shell console.
+
+vioapic
+=======
+
+"vioapic <vm_id>" shows the virtual IOAPIC information for a specific
+VM. In the following figure, we show the virtual IOPIC information for
+VM1:
+
+.. figure:: images/shell_image6.png
+   :align: center
+
+   vioapic information
+
+dump_ioapic
+===========
+
+"dump_ioapic" provides IOAPIC information and we can get IRQ number,
+IRQ vector number, etc.
+
+.. figure:: images/shell_image14.png
+   :align: center
+
+   dump_ioapic information
+
+pt
+==
+
+"pt" provides pass-through detailed information, such as the virtual
+machine number, interrupt type, interrupt request, interrupt vector,
+trigger mode, etc.
+
+.. figure:: images/shell_image13.png
+   :align: center
+
+   pt information
+
+int
+===
+
+"int" provides interrupt information on all CPUs and their corresponding
+interrupt vector.
+
+.. figure:: images/shell_image17.png
+   :align: center
+
+   int information
+
+cpuid
+=====
+
+"cpuid cpu_num" provides the highest EAX call parameter (leaf)
```

Comment:
`cpuid` is documented as `cpuid <leaf> [subleaf]`, is this `cpu_num` a new parameter?

Reply:
sorry, i forget to modify it, then modify like the following:
``cpuid <leaf> [subleaf]`` provides the CPUID leaf [subleaf] in hexadecimal.
![image](https://user-images.githubusercontent.com/3363158/55723539-e3077d80-5a3b-11e9-93be-73f49d832efb.png)
cpuid information